Lib/test/lock_tests.py

```diff
@@ -265,6 +271,25 @@ def test_state_after_timeout(self):
         self.assertFalse(lock.locked())
         self.assertTrue(lock.acquire(blocking=False))
 
+    @requires_fork
+    def test_at_fork_reinit(self):
+        def use_lock(lock):
+            # make sure that the lock still works normally
+            # after _at_fork_reinit()
+            lock.acquire()
+            lock.release()
+
+        # unlocked
+        lock = self.locktype()
+        lock._at_fork_reinit()
+        use_lock(lock)
+
+        # locked: _at_fork_reinit() resets the lock to the unlocked state
+        lock2 = self.locktype()
+        lock2.acquire()
+        lock2._at_fork_reinit()
+        use_lock(lock2)
```

Comment:
Hum, regrtest may complain that this test leaks memory. Maybe it should be run in a subprocess to ignore the leak.

Reply:
Hum no, it's ok. I tested: since PyMem_RawMalloc is used, regrtest doesn't bother us.